sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -370,7 +390,7 @@ case class Literal (value: Any, dataType: DataType) extends LeafExpression {
         case _ if v.isNaN => "'NaN'"
         case Float.PositiveInfinity => "'Infinity'"
         case Float.NegativeInfinity => "'-Infinity'"
-        case _ => v
+        case _ => s"'$v'"
```

Comment:
No big deal but it would be just `val castedValue = s"'$v'"`

```
scala> s"'${Double.NaN}'"
res0: String = 'NaN'

scala> s"'${Float.NegativeInfinity}'"
res1: String = '-Infinity'

scala> s"'${Float.PositiveInfinity}'"
res2: String = 'Infinity'
```

Reply:
ah good catch!

Reply:
oh, that's smarter..